registry/strings/functions.go

```diff
@@ -682,6 +682,52 @@ func (sr *StringsRegistry) SwapCase(str string) string {
 	}, str)
 }
 
+// Capitalize capitalizes the first letter of 'str'.
+//
+// Parameters:
+//
+//	str string - the string to capitalize.
+//
+// Returns:
+//
+//	string - the string with the first letter capitalized.
+//
+// Example:
+//
+//	{{ "hello world" | capitalize }} // Output: "Hello world"
+func (sr *StringsRegistry) Capitalize(str string) string {
+	for i, r := range str {
+		if unicode.IsLetter(r) {
+			return str[:i] + string(unicode.ToUpper(r)) + str[i+1:]
```

Comment:
I'm a bit afraid this logic might cause a lot of problems with Unicode

Please add tests about using

- `été`
- `👩🏾‍🦽foo`
- arabic characters

Maybe your code work, but the +1 let me think it doesn't

I think you have to play with runes

I'm not sure about the syntax


```suggestion
	for i, r := range runes(str) {
		if unicode.IsLetter(r) {
			return str[:i] + string(unicode.ToUpper(r)) + str[i+1:]
```

Look at the code of golang/x/text/case

https://cs.opensource.google/go/x/text/+/master:cases/map.go;l=59?q=Makeupper&ss=go%2Fx%2Ftext

It's not obvious

Reply:
Good catch, I will test to ensure that

Reply:
Following pull request : https://github.com/go-sprout/sprout/pull/63